integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedPruneRDD.scala

```diff
@@ -94,12 +90,13 @@ class DataMapPruneRDD(
         !finished
       }
 
-      override def next(): ExtendedBlocklet = {
+      override def next(): (String, ExtendedBlocklet) = {
         if (!hasNext) {
           throw new java.util.NoSuchElementException("End of stream")
         }
         havePair = false
-        val value = reader.getCurrentValue
+        val executorIP = SparkEnv.get.blockManager.blockManagerId.host
+        val value = (executorIP + "_" + cacheSize.toString, reader.getCurrentValue)
```

Comment:
please make sure that you do lastindexof while retrieving the ip and size

Reply:
done